File: bos_web/src/main/webapp/js/service/typeService.js
app.service('typeService',function($http){
    this.getTypeByPid =function (pid) {
        return $http.get('../type/getTypeByPid?pid=' + pid);
    }

    this.add = function (type) {
        return $http.post('../type/addType',type);
    }

    this.update = function (type) {
        return $http.post('../type/updateType',type);
    }
    this.findOne = function (id) {
        return $http.get('../type/findOne?id=' + id);
    }
    //删除
    this.dele=function(ids){
        return $http.get('../type/delete?ids='+ids);
    }
})

File: bos_service/src/main/java/com/bos/service/TypeService.java
package com.bos.service;

import com.bos.pojo.Type;

import java.util.List;

public interface TypeService {
    /**
     * 根据父id查询类型
     * @param pid
     * @return
     */
    public List<Type> getTypeByPid(Long pid);

    /**
     * 查询所有的种类
     * @return
     */
    public String[] findTypeArray();

    public int addType(Type type);

    public Type findOne(Long id);

    public int updateType(Type type);

    public void delete(Long[] ids);
}
